### PR TITLE
Refactor Unit Testing

### PR DESCRIPTION
**Pull request by mattBrzezinski**
## Overview

This CR refactors the current state of testing for `HTTP.jl`, pushing towards using `@testsets` for like-wise tests. It's quite a lengthy CR so it has been broken apart into multiple commits each one related to the refactor of a given component to make reviewing easier.

There were a few test files never run; `body.jl`, `mwe.jl`, and `uri_benchmark.jl`. I have included them in the `run_tests.jl` so they are not forgotten, however in their current state they are broken (see below for failures). I'll log issues for each of these, but their `@testsets` are marked with `@test_skip` to keep the CI happy.

## References
- `test/body.jl` failure -- #466 
- `test/mwe.jl` failure -- #467
- `test/uri_benchmark.jl` failure -- #468

### CODE REVIEW

File: test/body.jl [hunk 1]
```diff
@@ -1,53 +1,58 @@
...
-
-    io = Base.BufferStream()
-    @async begin
-        write(io, "Hello")
-        sleep(0.1)
-        write(io, "!")
-        sleep(0.1)
+@test_skip @testset "HTTP.Bodies" begin
+    @testset "Body" begin
+        @test String(take!(Body("Hello!"))) == "Hello!"
```
Comment (quinnj) [inline test/body.jl]: I think `Body` can be changed to `HTTP.bodybytes`; though these tests are pretty old and crufty, so I'm not sure if they're still useful (i.e. we might just remove them). But if you change to `HTTP.bodybytes`, and they work, we can keep them.

File: test/cookies.jl [hunk 2]
```diff
@@ -8,123 +8,129 @@
...
-        (Dict("Cookie" => "Cookie-1=v\$1; c2=v2"), "", [HTTP.Cookie("Cookie-1", "v\$1"), HTTP.Cookie("c2", "v2")]),
-        (Dict("Cookie" => "Cookie-1=v\$1; c2=v2"), "c2", [HTTP.Cookie("c2", "v2")]),
-        (Dict("Cookie" => "Cookie-1=\"v\$1\"; c2=\"v2\""), "", [HTTP.Cookie("Cookie-1", "v\$1"), HTTP.Cookie("c2", "v2")]),
-    ]
-    for (h, filter, cookies) in testcookies
-        @test HTTP.Cookies.readcookies(h, filter) == cookies
-    end
-end
-end; # @testset "Cookies"
+		(HTTP.Cookie("cookie-1", "v\$1"), "cookie-1=v\$1"),
```
Comment (quinnj) [inline test/cookies.jl]: why the indentation change here?

File: test/messages.jl [hunk 3]
```diff
@@ -17,167 +15,147 @@ using HTTP.MessageRequest: unknown_length
...
+        @test bodylength([SubString("Hello", 1,5), " ", SubString("World!",1,5)]) == 11
+        @test bodylength([UInt8[1,2,3], UInt8[4,5,6]]) == 6
+        @test bodylength([UInt8[1,2,3], view(UInt8[4,5,6],1:2)]) == 5
+        @test bodylength([view(UInt8[1,2,3],1:2), view(UInt8[4,5,6],1:2)]) == 4
+        @test bodylength(IOBuffer("foo")) == 3
+        @test bodylength([IOBuffer("foo"), IOBuffer("bar")]) == 6
+    end
 
+    @testset "Body Bytes" begin
+        @test bodybytes(7) == UInt8[]
```
Comment (quinnj) [inline test/messages.jl]: Looks like we already have test for `bodybytes`, so I say we just remove the `Body.jl` file.

File: test/uri_benchmark.jl [hunk 4]
```diff
@@ -1,45 +1,45 @@
+using HTTP
 using Unitful
 
-using HTTP
+include("resources/urlparser.jl")
```
Comment (quinnj) [inline test/uri_benchmark.jl]: this file wasn't really ever meant to be tests, so maybe just remove it, or move it to a top level `benchmark/` directory

**Comment by quinnj** [comment]
Another change that would be good is to switch from using AppVeyor to doing windows testing on travis, see the CSV.jl travis file for reference: https://github.com/JuliaData/CSV.jl/blob/master/.travis.yml.

File: test/cookies.jl (hunk 2 above)
Comment (mattBrzezinski) [reply test/cookies.jl]: Looks like this happened accidentally, also this file is using tabs. Fixing this now.

File: test/body.jl (hunk 1 above)
Comment (mattBrzezinski) [reply test/body.jl]: Looking more into it, `bodybytes` are being tested in `test/messages.jl`. This file can be removed.

**Comment by mattBrzezinski** [comment]
@quinnj TravisCI is passing now, can you remove the AppVeyor configuration for this project? I assume I don't have the perms to do that. I removed the `appveyor.yml` config file in this CR.

**Comment by quinnj** [comment]
Deleted. Thanks!

**Pull request merged**